### ISSUE TITLE
Crash on double-clicking the ColourPicker's titlebar

### ISSUE THREAD

**Issue by Ident** [cegui/cegui#1165]
**[Original report](https://bitbucket.org/cegui/cegui/issue/1165) by Nguyen Ngoc Huy (Bitbucket: [Nguyen Ngoc Huy](https://bitbucket.org/Nguyen Ngoc Huy), ).**

----------------------------------------

**Operating System:** Windows 10 (64-bit)

**OS version/build:** N/A

**GPU:** GeForce GTX 1050 Ti/PCIe/SSE2

**GPU drivers:** 398.36

The CEGUI samples framework crashes at the common dialogs example, in which clicking the titlebar of the ColourPicker widget would trigger the crash.

![](https://i.imgur.com/GC7x6OS.png)

Relevant stacktrace:
```
Program received signal SIGSEGV, Segmentation fault.
0x68941b30 in libCEGUICommonDialogs-0!_ZN5CEGUI12ColourPicker9setColourERKNS_6ColourE ()
   from C:\CEGUI\bin\libCEGUICommonDialogs-0.dll
(gdb) bt full
#0  0x68941b30 in libCEGUICommonDialogs-0!_ZN5CEGUI12ColourPicker9setColourERKNS_6ColourE ()
   from C:\CEGUI\bin\libCEGUICommonDialogs-0.dll
No symbol table info available.
#1  0x68943620 in libCEGUICommonDialogs-0!_ZN5CEGUI20ColourPickerControls21onAcceptButtonClickedERNS_15WindowEventArgsE () from C:\CEGUI\bin\libCEGUICommonDialogs-0.dll
No symbol table info available.
#2  0x641e1b1c in libCEGUIBase-0!_ZN5CEGUI11FrameWindow12toggleRollupEv ()
   from C:\CEGUI\bin\libCEGUIBase-0.dll
No symbol table info available.
#3  0x64240df9 in libCEGUIBase-0!_ZN5CEGUI8Titlebar20onMouseDoubleClickedERNS_14MouseEventArgsE ()
   from C:\CEGUI\bin\libCEGUIBase-0.dll
No symbol table info available.
#4  0x64175d43 in libCEGUIBase-0!_ZN5CEGUI10GUIContext21injectMouseButtonDownENS_11MouseButtonE ()
   from C:\CEGUI\bin\libCEGUIBase-0.dll
No symbol table info available.
#5  0x004107b2 in ?? ()
No symbol table info available.
#6  0x65b03f8b in libglfw!glfwSleep@8 () from C:\CEGUI\bin\libglfw.dll
No symbol table info available.
#7  0x65b07ec9 in libglfw!glfwWaitEvents@0 () from C:\CEGUI\bin\libglfw.dll
No symbol table info available.
#8  0x7677d313 in USER32!SetManipulationInputTarget () from C:\Windows\System32\user32.dll
No symbol table info available.
#9  0x7675e8da in USER32!DispatchMessageW () from C:\Windows\System32\user32.dll
No symbol table info available.
#10 0x7675df67 in USER32!CallWindowProcW () from C:\Windows\System32\user32.dll
No symbol table info available.
#11 0x7311ce43 in glDebugEntry () from C:\Windows\SYSTEM32\opengl32.dll
No symbol table info available.
```

It seems like double-clicking the titlebar would trigger `toggleRollup()`, which would somehow cause `ColourPicker::setColour()` to be called, thus causing the segfault.

**Reproducibility:** Always

**Reproduction steps:**

1. Compile CEGUI with `CEGUI_SAMPLES_ENABLED=1`

2. Open the ColourPicker widget from the samples framework.

3. Double-click the widget's titlebar.

**Comment by niello** [comment]
Not reproduced in master

**Comment by niello** [comment]
I don't believe that someone is going to fix bugs in 0.8.7, neither do I. This is fixed in master and I recommend to switch to it because now it is quite stable.
